crates/bitcoind_rpc/tests/test_emitter.rs

```diff
@@ -0,0 +1,300 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use bdk_bitcoind_rpc::Emitter;
+use bdk_chain::{
+    bitcoin::{Address, Amount, BlockHash, Txid},
+    local_chain::LocalChain,
+    Append, BlockId, ConfirmationHeightAnchor, IndexedTxGraph, SpkTxOutIndex,
+};
+use bitcoincore_rpc::RpcApi;
+
+struct TestEnv {
+    #[allow(dead_code)]
+    daemon: bitcoind::BitcoinD,
+    client: bitcoincore_rpc::Client,
+}
+
+impl TestEnv {
+    fn new() -> anyhow::Result<Self> {
+        let daemon = match std::env::var_os("TEST_BITCOIND") {
+            Some(bitcoind_path) => bitcoind::BitcoinD::new(bitcoind_path),
+            None => bitcoind::BitcoinD::from_downloaded(),
+        }?;
+        let client = bitcoincore_rpc::Client::new(
+            &daemon.rpc_url(),
+            bitcoincore_rpc::Auth::CookieFile(daemon.params.cookie_file.clone()),
+        )?;
```

Comment:
This works ok but I was wondering if there are reasons not to use
```
 // use BITCOIND_EXE instead of TEST_BITCOIND, see https://docs.rs/bitcoind/0.33.0/bitcoind/fn.exe_path.html
let daemon = bitcoind::BitcoinD::new(bitcoind::exe_path()?)?; 
```
and then using the inner `daemon.client` instead of creating a new one

Reply:
I think that will work with the latest version of `bitcoind`. However, we were using the old version before, where there was a custom package name for the `bitcoincore-rpc` dependency. Will change soon!

Reply:
Updated in c87b1b08bbfe63c22961093572b16828d9c355fe

Reply:
I was able to use to the latest version of `electrsd` (which depends on latest 0.33 `bitcoind`) without breaking MSRV with this commit: https://github.com/bitcoindevkit/bdk/pull/1110/commits/7be6f86343e30c9e07bcfd52342a203ea7556026

So I think we should be able to keep latest `bitcoind`.

Reply:
I have upgraded to the latest `bitcoind` crate. However, I think it is best to keep the implementation as is, just in case `bitcoincore_rpc` versions diverges between `bitcoind` and our dependencies?